src/EditorFeatures/Core/LanguageServer/AlwaysActiveLanguageClientEventListener.cs

```diff
@@ -60,8 +60,9 @@ private async Task LoadAsync()
                 using var token = _asynchronousOperationListener.BeginAsyncOperation(nameof(LoadAsync));
 
                 // Explicitly switch to the bg so that if this causes any expensive work (like mef loads) it 
-                // doesn't block the UI thread.
-                await TaskScheduler.Default;
+                // doesn't block the UI thread. Note, we always yield because sometimes our caller starts
+                // on the threadpool thread but is indirectly blocked on by the UI thread.
+                await TaskScheduler.Default.SwitchTo(alwaysYield: true);
```

Comment:
@sharwell should we just make this code the pattern we always use?  (and/or have a helper we ourselves use that always does this).  I'd like to ensure we aren't just patching this same issue oover and over again in multiple places.

Reply:
I guess I'm not sure why we didn't just Task.Run() in the StartListening call here...

Reply:
> should we just make this code the pattern we always use

No, that would be a waste. `await TaskScheduler.Default` is a safe and clear way to ensure the asynchronous operation is on a background thread, and NOP if you already are.